hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -318,6 +320,55 @@ bool relocatePCRelative(Env& env, Instruction* src, Instruction* dest,
         destCount += (env.destBlock.frontier() - destAddr) / kInstructionSize;
         isRelative = false;
       }
+    } else if (src->IsCompareBranch()) {
+      imm >>= vixl::kInstructionSizeLog2;
+      if (!is_int19(imm) || env.far.count(src)) {
```

Comment:
It would definitely be good to get this refactored a bit (and I suspect an extra decode won't be a big deal), but doing it as a separate PR seems perfectly fine to me.  This has been in rebase hell long enough (which is my fault)—let's get it landed :]